src/containerapp/azext_containerapp/custom.py

```diff
@@ -1510,6 +1519,963 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
+                            startup_command=None,
+                            args=None,
+                            scale_rule_metadata=None,
+                            scale_rule_name=None,
+                            scale_rule_type=None,
+                            scale_rule_auth=None,
+                            polling_interval=None,
+                            min_executions=None,
+                            max_executions=None,
+                            tags=None,
+                            no_wait=False,
+                            system_assigned=False,
+                            disable_warnings=False,
+                            user_assigned=None,
+                            registry_identity=None,
+                            workload_profile_name=None):
+    register_provider_if_needed(cmd, CONTAINER_APPS_RP)
+    validate_container_app_name(name, AppType.ContainerAppJob.name)
+    validate_create(registry_identity, registry_pass, registry_user, registry_server, no_wait)
+
+    if registry_identity and not is_registry_msi_system(registry_identity):
+        logger.info("Creating an acrpull role assignment for the registry identity")
+        create_acrpull_role_assignment(cmd, registry_server, registry_identity, skip_error=True)
+
+    if yaml:
+        if image or managed_env or trigger_type or replica_timeout or replica_retry_limit or\
+            replica_completion_count or parallelism or cron_expression or cpu or memory or registry_server or\
+            registry_user or registry_pass or secrets or env_vars or\
+                startup_command or args or tags:
+            not disable_warnings and logger.warning('Additional flags were passed along with --yaml. These flags will be ignored, and the configuration defined in the yaml will be used instead')
+        return create_containerappsjob_yaml(cmd=cmd, name=name, resource_group_name=resource_group_name, file_name=yaml, no_wait=no_wait)
+
+    if replica_timeout is None:
+        raise RequiredArgumentMissingError('Usage error: --replica_timeout is required')
+
+    if replica_retry_limit is None:
+        raise RequiredArgumentMissingError('Usage error: --replica_retry_limit is required')
+
+    if not image:
+        image = HELLO_WORLD_IMAGE
+
+    if managed_env is None:
+        raise RequiredArgumentMissingError('Usage error: --environment is required if not using --yaml')
+
+    # Validate managed environment
+    parsed_managed_env = parse_resource_id(managed_env)
+    managed_env_name = parsed_managed_env['name']
+    managed_env_rg = parsed_managed_env['resource_group']
+    managed_env_info = None
+
+    try:
+        managed_env_info = ManagedEnvironmentClient.show(cmd=cmd, resource_group_name=managed_env_rg, name=managed_env_name)
+    except:
+        pass
+
+    if not managed_env_info:
+        raise ValidationError("The environment '{}' does not exist. Specify a valid environment".format(managed_env))
```

Comment:
If you want to check env exist, only when exception is 404NotFound, you can sat it doesn't exist. For other errors, there may be other issues.

Reply:
this is similar to how we check in Container Apps, but I'll address it the next PR.